client/src/components/BlockEditor/types.tsx

```diff
@@ -4,3 +4,25 @@ export enum MutationStatus {
   Complete,
   Error
 }
+
+export interface Change {
+  value: any;
+  collapseToEnd(): Change;
+  collapseToStart(): Change;
+  insertTextByKey(key: any, offset: number, text: string): Change;
+  move(n: number): Change;
+  moveToRangeOf(node: any): Change;
+  removeTextByKey(key: string, offset: number, length: number): Change;
+}
+
+export interface TextNode {
+  key: string;
+  object: string;
+  leaves: any[];
+  text: string;
+}
+
+export interface Value {
+  document: any;
+  selection: any;
+}
```

Comment:
Kudos for adding the types here. Do you think we could take some of these from Definitely Typed instead, to keep it up to date with the Slate lib?

https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/slate